main.js

```diff
@@ -77,6 +86,17 @@ async function downloadCLI(version, platform, arch) {
   }
 }
 
+// Verifies the download attestation
+async function verifyDownloadAttestation(downloadPath) {
+  await exec.exec("gh", [
+    "attestation",
+    "verify",
+    downloadPath,
+    "--repo",
+    "UpCloudLtd/upcloud-cli",
+  ]);
```

Comment:
Hmph, https://github.com/UpCloudLtd/upcloud-cli-action/actions/runs/14857734295/job/41714932154?pr=3#step:8:10
>  gh: To use GitHub CLI in a GitHub Actions workflow, set the GH_TOKEN environment variable. Example:
>  env:
>    GH_TOKEN: ${{ github.token }}

Reply:
Maybe better to default to false then and add a note about this required env var in the docs for those who want to enable the verification 🤔

Reply:
4184bc75cb77374d5a4fea04c53c8a5771a1b1a0 takes a bit different approach -- as `GH_TOKEN` is required anyway, I think we can make use of it for toggling the verification.